src/components/data-display/tables/TableView.tsx

```diff
@@ -227,20 +228,116 @@ const TableView = <T, S>({
                 />
               </span>
             </button>
-            {Array.from({ length: table.getPageCount() }, (_, i) => (
-              <button
-                key={i}
-                onClick={() => table.setPageIndex(i)}
-                disabled={i === table.getState().pagination.pageIndex}
-                className={`w-[32px] h-[32px] items-center justify-center rounded text-sm ${
-                  i === table.getState().pagination.pageIndex
-                    ? 'bg-dark-grey text-white'
-                    : 'text-grey'
-                }`}
-              >
-                {i + 1}
-              </button>
-            ))}
+
+            {table.getPageCount() <= 6 ? (
```

Comment:
lot of logic for page selector control, would be good to refactor to a separate component in the future.